src/sp/it/pl/util/system/Os.kt

```diff
@@ -45,20 +45,22 @@ enum class Os {
     companion object {
 
         /** @return the current operating system */
-        @JvmStatic val current: Os
-            get() {
-                val osName = System.getProperty("os.name")
-                if (osName.startsWith("Windows")) return WINDOWS
-                if (osName.startsWith("Mac")) return OSX
-                if (osName.startsWith("SunOS")) return UNIX
-                if (osName.startsWith("Linux")) {
-                    val javafxPlatform = System.getProperty("javafx.platform")
-                    if (!("android"==javafxPlatform || "Dalvik"==System.getProperty("java.vm.name")))  // if not Android
-                        return UNIX
-                }
-                return UNKNOWN
+        @JvmStatic val current: Os = run {
+            val osName = System.getProperty("os.name")
+            when {
+                osName.indexOf("win")!=-1 -> WINDOWS
```

Comment:
1 why does it call run {} ?
2 Does your solution of Os detection provide any edge over the one that has been implemented already (besides the run {}) , the indexOf!=-1.
3 I also think >-1 is better than !=-1.

Reply:
1. Now it will be a constant variable instead of a getter. Thus it won't be recalculated whenever it's used, and why would it?
I just used `run` because you have to wrap a code block in something.
2. I wrote a pretty concise commit message for this:
```
Expand Os detection

Mostly modeled after uk.co.caprica.vlcj.runtime.RuntimeUtil
Now it will also detect FreeBSD and other Unixes as UNIX
```
3. ^copied from RuntimeUtil

Reply:
2 But why is it better? My solution was also inspired by someone elses code (I didnt come up with it). We should use the better one of course. Im simply not sure which one is.
3 That is not an argument.

Reply:
2. Because it takes more Unixes into account, as mentioned in the commit message
3. Well whether you use `>` or `!=` is a pure taste thing, because there is never a logical difference. I changed it back to `>` in the PR ;)

Reply:
2 I see. My bad. That escaped my attention (I blame being overworked).
3 Id argue its not taste thing. It is in this particular case but > is still more robust than != here. We merely know noone will change the implementation, but so what if we know? Some other times we wont and it is good to be consistent as well as good practice to form habit of practicing other good practices.